.github/workflows/python-release.yml

```diff
@@ -147,23 +147,23 @@ jobs:
         run: python3 -m poetry build --format=sdist
         if: startsWith(matrix.os, 'ubuntu')
 
+      - uses: astral-sh/setup-uv@v4 # required to use `uv` in cibuildwheel
+
       - name: Build wheels
         uses: pypa/cibuildwheel@v2.22.0
         with:
           output-dir: wheelhouse
           config-file: "pyproject.toml"
         env:
+          CIBW_BUILD_FRONTEND: "build[uv]"
           # Ignore 32 bit architectures
           CIBW_ARCHS: "auto64"
           CIBW_PROJECT_REQUIRES_PYTHON: ">=3.9,<3.13"
           CIBW_TEST_REQUIRES: "pytest==7.4.2 moto==5.0.1"
-          CIBW_TEST_EXTRAS: "s3fs,glue"
           CIBW_TEST_COMMAND: "pytest {project}/tests/avro/test_decoder.py"
-          # There is an upstream issue with installing on MacOSX
-          # https://github.com/pypa/cibuildwheel/issues/1603
           # Ignore tests for pypy since not all dependencies are compiled for it
           # and would require a local rust build chain
-          CIBW_TEST_SKIP: "pp* *macosx*"
```

Comment:
Nice! I also noticed this when reviewing https://github.com/apache/iceberg-python/pull/1391. Should we remove that one on `main` apart from this PR?? I think that would be something that we want to include in 0.9.0 anyway